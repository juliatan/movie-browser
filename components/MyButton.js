import React from 'react'
import {Button} from 'react-native'
import PropTypes from 'prop-types'

const MyButton = props => <Button title="test" onPress={() => {}} color={props.color || 'green'} />

MyButton.propTypes = {
  color: PropTypes.string,
}

export default MyButton
