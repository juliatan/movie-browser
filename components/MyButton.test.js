import React from 'react'
import {Button} from 'react-native'
import renderer from 'react-test-renderer'

import MyButton from './MyButton'

const getUnderlyingButton = testInstance => testInstance.root.findByType(Button)

describe('MyButton', () => {
  it('renders', () => {
    const button = renderer.create(<MyButton />).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('correctly overrides default color', () => {
    const color = 'red'
    const button = getUnderlyingButton(renderer.create(<MyButton color={color} />))
    expect(button.props.color).toBe(color)

    const color2 = 'blue'
    const button2 = getUnderlyingButton(renderer.create(<MyButton color={color2} />))
    expect(button2.props.color).toBe(color2)
  })
})
