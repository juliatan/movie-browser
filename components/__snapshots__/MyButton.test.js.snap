// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton renders 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={Object {}}
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "green",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
